soccer/planning/InterpolatedPath.hpp

```diff
@@ -80,22 +84,26 @@ namespace Planning
 			void startFrom(const Geometry2d::Point& pt, Planning::InterpolatedPath& result) const;
 
 			//Returns the destination of this path (the last point in the points array)
-			virtual boost::optional<Geometry2d::Point> destination() const;
+			virtual boost::optional<Geometry2d::Point> destination() const override;
 
 			/**
-			 * Returns true if the path never touches an obstacle or additionally, when exitObstacles is true, if the path
-			 * starts out in an obstacle but leaves and never re-enters any obstacle.
+			 * Returns true if the path hits an obstacle
 			 *
 			 * @param[in]	shape The obstacles on the field
-			 * @param[in] 	start The point on the path to start checking from
-			 * @return 		true if the path is valid, false if it hits an obstacle
+			 * @param[in] 	startTime The time on the path to start checking from
+			 * @return 		true if it hits an obstacle, otherwise false
 			 */
-			virtual bool hit(const Geometry2d::CompositeShape &shape, Geometry2d::Point *startPosition = NULL) const;
+			virtual bool hit(const Geometry2d::CompositeShape &shape, float startTime) const override;
 
-			// Set of points in the path - used as waypoints
-			std::vector<Geometry2d::Point> points;
-			std::vector<Geometry2d::Point> vels;
-			std::vector<float> times;
+			/**
+			 * Returns a subPath
+			 *
+			 * @param[in]	startTime The startTime for from which the subPath should be taken.
+			 * @param[in] 	endTime The endTime from which the subPath should be taken. If it is greater than the duration fo the path,
+								 it should go to the end of the path.
+			 * @return 	A unique_ptr to the new subPath
+			 */
+			virtual std::unique_ptr<Path> subPath(float startTime = 0, float endTime = std::numeric_limits<float>::infinity()) const override;
 
 			/**
 			 * Returns true if the path never touches an obstacle or additionally, when exitObstacles is true, if the path
```

Comment:
I guess thats a good point. Maybe I'll remove the subclass documentation.

On Tue, Jun 16, 2015 at 7:06 PM, Justin Buchanan <notifications@github.com>
wrote:

> In soccer/planning/InterpolatedPath.hpp
> <https://github.com/RoboJackets/robocup-software/pull/280#discussion_r32590302>
> :
>
> >
> > -			// Set of points in the path - used as waypoints
> > -			std::vector<Geometry2d::Point> points;
> > -			std::vector<Geometry2d::Point> vels;
> > -			std::vector<float> times;
> > +			/**
> > +			 * Returns a subPath
> > +			 *
> > +			 * @param[in]	startTime The startTime for from which the subPath should be taken.
> > +			 * @param[in] 	endTime The endTime from which the subPath should be taken. If it is greater than the duration fo the path,
> > +								 it should go to the end of the path.
> > +			 * @return 	A unique_ptr to the new subPath
> > +			 */
> > +			virtual std::unique_ptr<Path> subPath(float startTime = 0, float endTime = std::numeric_limits<float>::infinity()) const override;
> >
> >  			/**
> >  			 * Returns true if the path never touches an obstacle or additionally, when exitObstacles is true, if the path
>
> You've got it marked as override, so I think it would be fine to leave it
> undocumented here and the reader can just look at the superclass method
> where it's declared for the first time and see the description. Any place
> where you copy comments, they're bound to get out of sync.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RoboJackets/robocup-software/pull/280/files#r32590302>
> .
>